dataset/balance_data.py

```diff
@@ -0,0 +1,83 @@
+import h5py
+import pandas as pd
+from utils.helpers import *
+
+def undersample(df):
+    """Undersamples data as per particle class 
+
+    Args:
+        df (pandas dataframe): Dataframe containing entries for a momentum bin
+
+    Returns:
+        pandas dataframe: Dataframe after undersampling entries
+    """
+    classes = df.label.value_counts().to_dict()
+    least_class_amount = min(classes.values())
+    classes_list = []
+    for key in classes:
+        classes_list.append(df[df['label'] == key]) 
+    classes_sample = []
+    for i in range(0,len(classes_list)-1):
+        classes_sample.append(classes_list[i].sample(least_class_amount))
+    df_maybe = pd.concat(classes_sample)
+    final_df = pd.concat([df_maybe,classes_list[-1]], axis=0)
+    final_df = final_df.reset_index(drop=True)
+    return final_df
+
+
+def balance_data(
+    dset_path=get_config("dataset.train").keys(), 
+    bdset_path=get_config("dataset.train").values()
+):
+    """Creates balanced dataset with undersampled momentum bins and saves it to specified path
+
+    Args:
+        dset_path (list, optional): list of source hdf5 file paths. Defaults to get_config("dataset.train").keys().
+        bdset_path (list, optional): list of destination hdf5 file paths. Defaults to get_config("dataset.train").values().
+    """
+
+    for file_path in dset_path:
+        f = h5py.File(file_path)
+        df=events_to_pandas(f)
+
+
+    # Reading source hdf5 file to pandas dataframe
```

Comment:
Should this code below be in the above `for` loop?

Reply:
Agreed and modifications done